### PR TITLE
Add "isPrimary" flag to job creation request

### LINKED ISSUE #60: Add the ability to create hidden jobs.
The is_primary flag should be initially set to false in this case.

### PR DESCRIPTION
**Pull request by Foxcapades**
resolves #60

### CODE REVIEW

File: rest-service/api/src/main/java/org/veupathdb/service/multiblast/service/http/job/JobCreator.java [hunk 1]
```diff
@@ -101,12 +101,8 @@ public void handleInitialRun(JobDetails d) throws Exception {
       Arrays.asList(d.targets)
     );
 
-    if (d.parentHash != null) {
-      var noLinks = db.getJobLinks(d.parentHash).stream()
-        .map(JobLink::jobHash)
-        .noneMatch(h -> Arrays.equals(d.hash, h));
-      if (noLinks)
-        db.createJobLink(d.hash, d.parentHash);
+    if (d.parentHash != null && !db.jobToJobLinkExists(d.parentHash, d.hash)) {
```
Comment (Foxcapades) [inline rest-service/api/src/main/java/org/veupathdb/service/multiblast/service/http/job/JobCreator.java:104]: Unrelated cleanup, idk why i decided it goes in this branch.

**Review by jtlong3rd — APPROVED** [review]
